include/seqan3/core/detail/pack_algorithm.hpp

```diff
@@ -0,0 +1,244 @@
+ * \returns `true` if the predicate returns `true` for each type in the type list, `false` otherwise.
+ *
+ * \details
+ *
+ * This function operates on types instead of values.
+ * The following steps are performed to call the passed predicate on the types contained in the type list:
+ *
+ *  * expand the types within the type list
+ *  * wrap each type in std::identity and instantiate it (i.e. transform a type into a value)
+ *  * call the parameter pack version of seqan3::detail::all_of with the instances of std::identity (as a pack).
+ *
+ * Note that wrapping the types in std::type_identity is a technical trick to make a type representable as a value.
+ * Instantiating a type might not work because they might not be std::default_constructible.
+ * In addition it is possible, to invoke the predicate on incomplete types.
+ *
+ * ### Example
+ *
+ * \include test/snippet/core/detail/all_of_in_type_list.cpp
+ *
+ * ### Complexity
+ *
+ * Linear in the number of types in the seqan3::type_list.
+ *
+ * [Compile-time complexity: Linear number of template instantiations.]
+ */
+template <typename type_list_t, typename unary_predicate_t>
+[[nodiscard]] constexpr bool all_of(unary_predicate_t && fn)
+//!\cond
+    requires type_list_specialisation<type_list_t>
+//!\endcond
+{
+    return type_list_expander<type_list_t>::invoke_on_type_identities([&] (auto && ...type_identities)
+    {
+        return all_of(fn, std::forward<decltype(type_identities)>(type_identities)...);
+    });
+}
+
+//-----------------------------------------------------------------------------
+// for_each
+//-----------------------------------------------------------------------------
+
+/*!\brief Applies a function to each element of an argument pack.
+ * \ingroup core
+ *
+ * \tparam unary_function_t The function type, like function pointers, functors and lambdas.
+ * \tparam pack_t The parameter pack of the arguments (each argument type can be different).
+ *
+ * \param[in] fn The function to call on every argument.
+ * \param[in] args The parameter pack.
+ *
+ * \details
+ *
+ * This function behaves like std::for_each but on parameter packs. The invocation(s) will be done without any loop.
+ *
+ * ### Example
+ *
+ * \include test/snippet/core/detail/for_each_in_pack.cpp
+ *
+ * ### Complexity
+ *
+ * Linear in the number of elements in the pack.
+ *
+ * \attention Opposed to the std::for_each the argument order is changed, such that the first argument is the unary
+ * function to invoke on each argument followed by the arguments.
+ * This is due to a constraint in the c++ language regarding parameter packs.
+ *
+ * \sa https://en.cppreference.com/w/cpp/language/parameter_pack
+ */
+template <typename unary_function_t, typename ...pack_t>
+//!\cond
+    requires (std::invocable<unary_function_t, pack_t> && ... && true)
+//!\endcond
+constexpr void for_each(unary_function_t && fn, pack_t && ...args)
+{
+    [[maybe_unused]] int r = (fn(std::forward<pack_t>(args)), ..., 0);
```

Comment:
Why do you need to store this inside an unused variable?
This should work also:
```suggestion
    (fn(std::forward<pack_t>(args)), ...);
```

Reply:
that works equally likely, yes